18.0 tutorial drod

vandroogenbd opened the pull request.

vandroogenbd committed:
[ADD] estate: Finish up to chapter 10 (included)

vandroogenbd committed:
[IMP] estate: Finish up to Chapter 12

vandroogenbd committed:
[FIX] estate: Fix reference errors in views

vandroogenbd committed:
[IMP] estate: Finish Chapter 12

vandroogenbd committed:
[ADD] estate: Add estate_account model, finish Chapter 13

vandroogenbd committed:
[IMP] estate: Add a (somewhat decent) kanban view & fix warnings

vandroogenbd committed:
[IMP] estate: Make runbot's style checking happy

vandroogenbd committed:
[ADD] awesome_owl: Finish Chapter 6

[ADD] awesome_owl: Add ugly-ass boxes (Chapter 3)

[IMP] awesome_owl: Finish Chapter 4

I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript
I hate JavaScript

[IMP] awesome_owl: Finish Chapter 5

[IMP] awesome_owl: Finish Chapter 6

proose commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,85 @@
+    @api.model_create_multi
+    def create(self, vals_list):
+        for record in vals_list:
+            _property = self.env["estate_property"].browse(record["property_id"])
```
Using an underscore before a variable is usually to indicate a private variable, or a value that won't be used (if we use a method that returns a tuple of values for example). Best not to use it here

proose commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,85 @@
+    def create(self, vals_list):
+        for record in vals_list:
+            _property = self.env["estate_property"].browse(record["property_id"])
+            if record["price"] < _property["best_offer"]:
```
```suggestion
            if record["price"] < _property.best_offer:
```
`self.env['model.name'].browse(SOMEID)` will return a recordset, so you can (and should preferably) get its fields like attributes

proose commented on estate/static/description/icon.webp:
Why and where is this file needed ?

proose commented on estate/views/estate_menu_views.xml:
```diff
@@ -0,0 +1,51 @@
+        name="Estate"
+    />
+
+    <!-- HEADER: Advertisements -->
```
You don't need these comments. If you don't find this way clear enough, you can nest your menuitems:
```
<menuitem id="coucou">
    <menuitem id="salut"/>
<menuitem/>
```

"salut" sera un sous-menuitem de "coucou"

proose commented on estate/models/estate_property_infos.py:
```diff
@@ -0,0 +1,52 @@
+
+    color = fields.Integer(string="Colour")
+
+    property_estate_ids = fields.Many2many(
```
This is unnecessary, we just want the tags on the properties, but we don't really care to find properties from a tag (a search by tags on the properties list view will suffice if we want to find properties with a special tag

proose commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,173 @@
+                    <group>
+                        <group>
+                            <field name="type_id" attributes="{'can_write': false}"
+                                options="{'no_create': true, 'no_open': true}" />
```
Why those options and attribute ?

proose commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,173 @@
+        <field name="name">estate_property_kanban</field>
+        <field name="model">estate_property</field>
+        <field name="arch" type="xml">
+            <kanban class="o_kanban_mobile" default_group_by="type_id">
```
Don't forget your kanban view shouldn't allow drag and drop

proose commented on estate_account/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+    'name': '[TUTO] Estate_Account',
+    'summary': 'Module to invoice sold real estate properties',
+    'depends': [
+        'base_setup', 'estate', 'account'
```
'base_setup' here is not needed, as it is already a parent of 'account' which this module depends on ;)

proose commented on estate_account/__manifest__.py:
```diff
@@ -0,0 +1,16 @@
+        'base_setup', 'estate', 'account'
+    ],
+    'license': "LGPL-3",
+    'data': [
```
No need to add empty list in the manifest

proose reviewed (commented):
Very very nice work ! You thought about (almost) *every* corner cases, well done !
You can quickly work on the different comments I sent you and squash your commits in one (only yours this time :smile: ).
Well done !

vandroogenbd committed:
[ADD] awesome_owl: Finish Chapter 7

[IMP] awesome_owl: Finish Chapter 8

[IMP] awesome_owl: Finish Chapter 9

[IMP] awesome_owl: Finish Chapter 10
